src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,153 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode;
+        if (first == null) {
+            newNode = new Node<>(null, value, null);
+            first = newNode;
+        } else {
+            newNode = new Node<>(last, value, null);
+            last.next = newNode;
+        }
```

Comment:
since `first` is `null` then `last` also `null`, no point to duplicate `new Node<>` just introduce `new Node<>(last, value, null);` at line 12

Reply:
done)